database/postgresql.py

```diff
@@ -224,3 +225,88 @@ def _unsplit_cnpj(self, cnpj_basico: str, cnpj_ordem: str, cnpj_dv: str) -> str:
             cnpj_ordem=str(cnpj_ordem).zfill(4),
             cnpj_dv=str(cnpj_dv).zfill(2),
         )
+
+class PostgreSQLDatabaseAggregates(AggregatesDatabaseInterface):
+    def __init__(self, host, database, user, password, port):
+        self.host = host
+        self.database = database
+        self.user = user
+        self.password = password
+        self.port = port
+        
+    def _select(self, command: str, data: Dict = {}) -> Iterable[Tuple]:
+        connection = psycopg2.connect(
+            dbname=self.database,
+            user=self.user,
+            password=self.password,
+            host=self.host,
+            port=self.port,
+        )
+        with connection.cursor() as cursor:
+            cursor.execute(command, data)
+            logging.debug(f"Starting query: {cursor.query}")
+            for entry in cursor:
+                logging.debug(entry)
+                yield entry
+            logging.debug(f"Finished query: {cursor.query}")
+    
+    def _always_str_or_none(self, data: Any) -> Union[str, None]:
+        if data == "None" or data == "" or data is None:
+            return None
+        elif not isinstance(data, str):
+            return str(data)
+        else:
+            return data
+        
+    def _format_aggregates_data(self, data: Tuple) -> Dict:
+        formatted_data = [self._always_str_or_none(value) for value in data]
+        return {
+            "territory_id": formatted_data[1],
+            "state_code": formatted_data[2],
+            "url_zip": formatted_data[3],
+            "year": formatted_data[4],
+            "last_updated": formatted_data[5],
+            "hash_info": formatted_data[6],
+            "file_size": formatted_data[7]
+        }
+
+    def get_aggregates(self, territory_id: Optional[str] = None, state_code: str = "") -> Union[List[Aggregates], None]:
+        if territory_id is None:
+            command = """
+                SELECT
+                    *
+                FROM
+                    aggregates
+                WHERE
+                    state_code = %(state_code)s
+                    AND
+                    territory_id IS NULL
+            """
+            data = {
+                "state_code": state_code
+            }
+        else:
+            command = """
+                SELECT
+                    *
+                FROM
+                    aggregates
+                WHERE
+                    territory_id = %(territory_id)s 
+                    AND 
+                    state_code = %(state_code)s
+            """
+            data = {
+                "territory_id": territory_id,
+                "state_code": state_code
+            }
```

Comment:
Só estou fazendo essa sugestão pois acho que a mudança entre um select e outro é muito simples. Normalmente sugeriria criar uma função específica pra cada comando pra gente ter uma anotação do que significa cada comando SQL no nome da função.
```suggestion
        command = """
            SELECT
                *
            FROM
                aggregates
            WHERE
                state_code = %(state_code)s
                AND
                territory_id {territory_id_query_statement}
        """
        data = {
            "state_code": state_code
        }
        if territory_id is None:
            command = command.format(territory_id_query_statement="IS NULL")
        else:
            data["territory_id"] = territory_id
            command = command.format(territory_id_query_statement="= %(territory_id)s")

```